packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -273,11 +273,11 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
         // in the case that an honest user proves their withdrawal against a dispute game that
         // resolves against the root claim, or the dispute game is blacklisted, we allow
         // re-proving the withdrawal against a new proposal.
-        IDisputeGame game = provenWithdrawal.disputeGameProxy;
+        IDisputeGame oldGame = provenWithdrawal.disputeGameProxy;
         require(
-            provenWithdrawal.timestamp == 0 || gameProxy.status() == GameStatus.CHALLENGER_WINS
-                || disputeGameBlacklist[gameProxy] || game.gameType().raw() != respectedGameType.raw(),
-            "OptimismPortal: withdrawal hash has already been proven, and dispute game is not invalid"
+            provenWithdrawal.timestamp == 0 || oldGame.status() == GameStatus.CHALLENGER_WINS
```

Comment:
Hmmm may have to `(provenWithdrawal.timestamp == 0 && address(oldGame) == address(0))` here, I assume this will throw if the withdrawal has never been proven before.

Reply:
Shouldn't with the evaluation being from L->R. If the withdrawal's never been proven before, this require will pass on the first condition before the `oldGame` is referenced with a call.